pkg/core/secrets/manager/validator.go

```diff
@@ -0,0 +1,72 @@
+package manager
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	mesh_proto "github.com/Kong/kuma/api/mesh/v1alpha1"
+	"github.com/Kong/kuma/pkg/core/ca"
+	mesh_core "github.com/Kong/kuma/pkg/core/resources/apis/mesh"
+	"github.com/Kong/kuma/pkg/core/resources/model"
+	core_store "github.com/Kong/kuma/pkg/core/resources/store"
+	"github.com/Kong/kuma/pkg/core/validators"
+)
+
+// MeshAccessor is used instead of MeshManager, otherwise we would have cyclic dependency.
+// MeshManager depends on SecretManager which depends on SecretValidator which would depend on ResourceManager
+type MeshAccessor = func(context.Context, model.Resource, ...core_store.GetOptionsFunc) error
+
+type SecretValidator interface {
+	ValidateDelete(ctx context.Context, secretName string, secretMesh string) error
+}
+
+func NewSecretValidator(caManagers ca.Managers, meshAccessor MeshAccessor) SecretValidator {
```

Comment:
Why we can't pass here ReadOnlyResourceManager?

Reply:
`ReadOnlyResourceManager` is eventually consistent

Reply:
Yes right, because of caches. What about global ResourceManager?

Reply:
I explained in the comment of `MeshAccessor` why we need this. Otherwise we have a cycle.

Reply:
Yes I read the comment, but ResourceManager as an interface doesn't depend on MeshManager, it is a high-level abstraction

Reply:
Can you try to send me a patch how would you see this being implemented? That would be helpful. I think I'm missing something.

Reply:
I'm pretty sure that I'm missing something :) But anyway here is a patch, will it work?
https://gist.github.com/lobkovilya/35fd97131f0fd1de7b251b34b4a00a39

Reply:
It won't work because here
```
+		validator := secret_manager.NewSecretValidator(builder.CaManagers(), builder.ResourceManager())
```
`builder.ResourceManager()` returns nil

Reply:
But we initialize it with `builder.WithResourceManager(customizableManager)` in the code before?

Reply:
Oh I watched at `plugin.go` but you apparently pointer do `bootstrap.go`

Reply:
That a bit strange, you capture `builder` object as a clojure for MeshAccessor function. Object `builder` should exist only during bootstrap process, probably we don't want to care what will happen with it after `builder.Build()`. 

SecretValidator is a part of the manager-layer, can we point it right to the ResourceStore?

Reply:
Seems a bit hacky to reuse ResourceStore. The idea for Manager is that any part of code that retrieve resources should you manager, not store (because of extra logic that might be build there). But ok, maybe it's a little bit hacky that exposing builder in lambda.

Changed.

Reply:
We can receive that ResourceStore as a ReadOnlyResourceManager interface. That will prevent the store from direct writes (bypassing the manager's logic). But that potentially might confuse further code-readers, why we pass store through the manager interface. 
So I'm fine with the current implementation.